lib/Doctrine/ORM/Repository/DefaultRepositoryFactory.php

```diff
@@ -28,11 +28,8 @@ public function getRepository(EntityManagerInterface $entityManager, $entityName
     {
         $repositoryHash = $entityManager->getClassMetadata($entityName)->getClassName() . spl_object_id($entityManager);
 
-        if (isset($this->repositoryList[$repositoryHash])) {
-            return $this->repositoryList[$repositoryHash];
-        }
-
-        return $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
+        return $this->repositoryList[$repositoryHash] ??
+            $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
```

Comment:
I don't like this change: it's easy to mess with operator precedence this way.

Reply:
This is quite standard

Reply:
Btw in this context I would prefer:
```php
if (!isset($this->repositoryList[$repositoryHash])) {
    $this->repositoryList[$repositoryHash] = $this->createRepository($entityManager, $entityName);
}

return $this->repositoryList[$repositoryHash];
```
So Setter and Getter parts are strictly detached

Reply:
I moved all my lazy getters to following:

```php
public function getFoo()
{
    return $this->foo
       ?? $this->foo = $this->makeFoo();
}
```

The intent is clearer and the expression is quite simple

Reply:
See for example https://github.com/Ocramius/ProxyManager/blob/81d53b2878f1d1c40ad27270e64b51798485dfc5/src/ProxyManager/Configuration.php :-)

Reply:
Ok then.

@carusogabriel may you move the `??` operator to the new line so conjunctions operators are easy to spot?

Reply:
Sure. I’m on city now. I’ll update my PR at Doctrine’s projects this afternoon 😄